pkg/sbom/spdx/marshal.go

```diff
@@ -148,11 +149,23 @@ func (m *Marshaler) Marshal(r types.Report) (*spdx.Document, error) {
 			if err != nil {
 				return nil, xerrors.Errorf("package file error: %w", err)
 			}
-			spdxFiles = append(spdxFiles, files...)
-			for _, file := range files {
-				relationShips = append(relationShips,
-					relationShip(spdxPackage.PackageSPDXIdentifier, file.FileSPDXIdentifier, RelationShipContains),
-				)
+			if files != nil {
+				spdxFiles = append(spdxFiles, files...)
+				for _, file := range files {
+					relationShips = append(relationShips,
+						relationShip(spdxPackage.PackageSPDXIdentifier, file.FileSPDXIdentifier, RelationShipContains),
+					)
+				}
+
+				verificationCode, err := spdxutils.GetVerificationCode(files, "")
+				if err != nil {
+					return nil, xerrors.Errorf("package verification error: %w", err)
+				}
+				// GetVerificationCode creates empty ExcludedFiles array
+				verificationCode.ExcludedFiles = nil
```

Comment:
created https://github.com/spdx/tools-golang/pull/230 for this case

Reply:
Looks like it was merged.

Reply:
Used commit with this change - https://github.com/aquasecurity/trivy/pull/5533/commits/b8fb207d3ea213662929e40042d82277c90743e9